ios/FluentUI/Vnext/Notification/NotificationView_SwiftUI.swift

```diff
@@ -45,41 +42,98 @@ import SwiftUI
+                dismissAction: (() -> Void)? = nil) {
+        let state = MSFNotificationStateImpl(style: style, message: message)
+        state.title = title
+        state.image = image
+        state.actionButtonTitle = actionButtonTitle
+        state.actionButtonAction = actionButtonAction
+        state.messageButtonAction = messageButtonAction
+        state.dismissAction = dismissAction
         self.state = state
-    }
-
-    private var hasImage: Bool {
-        state.style.isToast && state.image != nil
-    }
 
-    private var hasSecondTextRow: Bool {
-        state.style.isToast && state.title != ""
+        if let isPresented = isPresented {
+            _isPresented = isPresented
+        } else {
+            _isPresented = .constant(true)
+        }
     }
 
-    private var hasCenteredText: Bool {
-        !state.style.isToast && state.actionButtonAction == nil
+    public var body: some View {
+        GeometryReader { geometryReader in
+            let width = geometryReader.size.width
+            innerContents
+                .onTapGesture {
+                    if let messageAction = state.messageButtonAction, let dismissAction = state.dismissAction {
+                        isPresented = false
+                        dismissAction()
+                        messageAction()
+                    }
+                }
+                .frame(width: state.style.isToast && horizontalSizeClass == .regular ? width / 2 : width - (2 * tokens.presentationOffset), alignment: .bottom)
+                .background(
+                    RoundedRectangle(cornerRadius: tokens.cornerRadius)
+                        .strokeBorder(Color(dynamicColor: tokens.outlineColor), lineWidth: tokens.outlineWidth)
+                        .background(
+                            RoundedRectangle(cornerRadius: tokens.cornerRadius)
+                                .fill(Color(dynamicColor: tokens.backgroundColor))
+                        )
+                        .shadow(color: Color(dynamicColor: tokens.ambientShadowColor),
+                                radius: tokens.ambientShadowBlur,
+                                x: tokens.ambientShadowOffsetX,
+                                y: tokens.ambientShadowOffsetY)
+                        .shadow(color: Color(dynamicColor: tokens.perimeterShadowColor),
+                                radius: tokens.perimeterShadowBlur,
+                                x: tokens.perimeterShadowOffsetX,
+                                y: tokens.perimeterShadowOffsetY)
+                )
+                .onChange(of: isPresented, perform: { present in
+                    if present {
+                        presentAnimated()
+                    } else {
+                        dismissAnimated()
+                    }
+                })
+                .padding(.bottom, tokens.bottomPresentationPadding)
+                .measureSize { viewSize in
```

Comment:
Once this is changed to **.onSizeChanged**, I believe **newSize** is a good name for the parameter of the closure (instead of viewSize).